backend/src/test/java/gov/cdc/usds/simplereport/service/PatientLinkServiceTest.java

```diff
@@ -32,26 +33,10 @@ void setupData() {
   @Test
   void getPatientLinkCurrent() throws Exception {
     Organization org = _organizationService.getCurrentOrganization();
-    Facility facility = _organizationService.getFacilities(org).get(0);
-    Person p =
-        _personService.addPatient(
-            null,
-            "FOO",
-            "Fred",
-            null,
-            "",
-            "Sr.",
-            LocalDate.of(1865, 12, 25),
-            _dataFactory.getAddress(),
-            "8883334444",
-            PersonRole.STAFF,
-            null,
-            null,
-            null,
-            null,
-            false,
-            false);
+    Facility facility = _dataFactory.createValidFacility(org);
+    Person p = _dataFactory.createFullPerson(org);
 
+    TestUserIdentities.addFacilityAuthorities(facility);
```

Comment:
This pattern seems ugly to me—I know it's the natural extension of the way I handled a similar problem last week, which also seemed ugly to me, so I'm not throwing any shade at the doing of what needed to be done in order to get across the finish line, but we should probably figure out better patterns or we're going to have a lot of unmaintainable test code in the future.

Reply:
i had thought about doing something akin to runWith() that takes an executable, but then i ran into the problem of having to run a _callable_ with a particular set of facility authorities, then a _callable that returned multiple objects_, and ultimately ended up with the paradigm that i needed to be able to just add and remove facility authoriities. i'm open to doing something a bit "cleaner" like the `updateSelfPrivileges()` in the API tests which erases the need to have separate add and remove methods, but beyond that, i think we'd need to substantially reenvision our test scaffolding